File: client/src/services/sellerOrderService.js
import api from './api';

const sellerOrderService = {
  getSellerOrders: async (page = 0, size = 10) => {
    const response = await api.get(`/api/seller/orders?page=${page}&size=${size}`);
    return response.data;
  },

  getSellerOrderById: async (id) => {
    const response = await api.get(`/api/seller/orders/${id}`);
    return response.data;
  },

  confirmOrder: async (id) => {
    const response = await api.put(`/api/seller/orders/${id}/confirm`);
    return response.data;
  },

  shipOrder: async (id) => {
    const response = await api.put(`/api/seller/orders/${id}/ship`);
    return response.data;
  },

  completeOrder: async (id) => {
    const response = await api.put(`/api/seller/orders/${id}/complete`);
    return response.data;
  }
};

export default sellerOrderService;

File: server/src/main/java/com/fptmarket/service/SellerOrderService.java
package com.fptmarket.service;

import com.fptmarket.dto.response.SellerOrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface SellerOrderService {
    Page<SellerOrderResponse> getSellerOrders(Pageable pageable);
    SellerOrderResponse getSellerOrderById(Long orderId);
    SellerOrderResponse confirmOrder(Long orderId);
    SellerOrderResponse shipOrder(Long orderId);
    SellerOrderResponse completeOrder(Long orderId);
}
